### PR TITLE
INTYGFV-15815: Implementation of FAKE-login and SAML-shell. Added Testability endpoint that supplies test data.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
See comments.. tried to review this in IntelliJ and it worked (or didnt work) :-)

File: integration-api/src/main/java/se/inera/intyg/minaintyg/integration/api/PUIntegrationService.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package se.inera.intyg.minaintyg.integration.api;
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply integration-api/src/main/java/se/inera/intyg/minaintyg/integration/api/PUIntegrationService.java:1]: Agree!

File: integration-pu/src/main/java/se/inera/intyg/minaintyg/integration/pu/PUIntegrationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package se.inera.intyg.minaintyg.integration.pu;
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply integration-pu/src/main/java/se/inera/intyg/minaintyg/integration/pu/PUIntegrationServiceImpl.java:1]: Agree!

File: integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/intygproxyservice/person/webcert/WebcertCertificateIntegrationService.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package se.inera.intyg.minaintyg.integration.intygproxyservice.person.webcert;
```
Comment (mww-sopra-steria) [inline integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/intygproxyservice/person/webcert/WebcertCertificateIntegrationService.java:1]: Change the package to something like: se.inera.intyg.minaintyg.integration.webcert (exact package can be adjusted in future pr:s when this is implemented.

File: settings.gradle [hunk 4]
```diff
@@ -2,4 +2,6 @@ rootProject.name = 'minaintyg'
 
 include 'web'
 include 'integration-api'
-include 'integration-webcert'
\ No newline at end of file
+include 'integration-webcert'
+include 'integration-pu'
```
Comment (mww-sopra-steria) [inline settings.gradle:6]: Should the module be called: integration-intyg-proxy-service?

File: web/build.gradle [hunk 5]
```diff
@@ -3,12 +3,22 @@ plugins {
 }
 
 dependencies {
+    constraints {
```
Comment (mww-sopra-steria) [inline web/build.gradle:6]: Let's look at this in the next PR concerning SAML integration with IdP. We should be careful to add specific versions in submodules.

File: web/src/test/java/se/inera/intyg/minaintyg/auth/MinaIntygLoggingSessionRegistryImplTest.java [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package se.inera.intyg.minaintyg.auth;
+
+class MinaIntygLoggingSessionRegistryImplTest {
```
Comment (mww-sopra-steria) [inline web/src/test/java/se/inera/intyg/minaintyg/auth/MinaIntygLoggingSessionRegistryImplTest.java:3]: Class needed? ;-)

File: build.gradle [hunk 7]
```diff
@@ -13,6 +13,9 @@ version = '4.0.0-SNAPSHOT'
 allprojects {
     repositories {
         mavenCentral()
+        maven { url "https://repo.spring.io/milestone" }
```
Comment (mww-sopra-steria) [inline build.gradle:16]: Are the repo.spring.io/milestone and snapshot needed? Worked locally for me without them, but the shibboleth repo was necessary. 

Lets look at SAML dependencies more later in a future PR.

**Review by mww-sopra-steria — APPROVED** [review]
Great work! Take a look at some very minor comments before merging!

File: build.gradle (hunk 7 above)
Comment (ChrilleSigerhed) [reply build.gradle]: Tested the implementation without these dependencies and it worked, so I removed them.

**Pull request merged**